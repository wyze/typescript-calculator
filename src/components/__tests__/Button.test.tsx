import { css } from 'glamor'
import { h } from 'preact'
import Button from '../Button'
import snapshot from './utils/snapshot'

const props = {
  action: jest.fn(),
  text: '.',
}

const propsWithClassName = {
  ...props,
  className: css({ color: 'rebeccapurple' }),
}

const propsWithHtmlEntities = {
  ...props,
  text: '&minus;',
}

describe('<Button />', () => {
  it('renders', () => snapshot(<Button {...props} />))
  it('renders with className', () => snapshot(<Button {...propsWithClassName} />))
  it('renders with html entities', () => snapshot(<Button {...propsWithHtmlEntities} />))
  xit('handles onclick event')
})
