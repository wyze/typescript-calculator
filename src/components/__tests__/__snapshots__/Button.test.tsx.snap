// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders 1`] = `
<button
  className="css-1ht4jl3"
  dangerouslySetInnerHTML={
    Object {
      "__html": ".",
    }
  }
  onClick={[Function]}
/>
`;

exports[`<Button /> renders with className 1`] = `
<button
  className="css-mtanlf"
  dangerouslySetInnerHTML={
    Object {
      "__html": ".",
    }
  }
  onClick={[Function]}
/>
`;

exports[`<Button /> renders with html entities 1`] = `
<button
  className="css-1ht4jl3"
  dangerouslySetInnerHTML={
    Object {
      "__html": "&minus;",
    }
  }
  onClick={[Function]}
/>
`;
